odl/util/testutils.py

```diff
@@ -284,7 +284,7 @@ def noise_array(space):
     """
     from odl.space import ProductSpace
     if isinstance(space, ProductSpace):
-        return np.array([noise_array(si) for si in space])
+        return np.array([noise_array(si) for si in space]).reshape(space.shape)
```

Comment:
I dont get it, this should fail, no? `space.shape` is usually a small number for product spaces but the array is very big.